### PR TITLE
gh-117381: Fix error message for `ntpath.commonpath`

### PR DESCRIPTION
**Pull request by nineteendo**
* Issue: gh-117381

### CODE REVIEW

**Comment by aisk** [comment]
I think maybe we need a test to ensure the expected error will be raised.

**Comment by nineteendo** [comment]
There are now `assertRaisesRegex` tests for all cases:

```mermaid
graph TD;
D:/foo---C:/foo
D:/foo---C:foo
D:foo---C:/foo
D:foo---C:foo
C:/foo---/foo
C:/foo---foo
C:foo---/foo
C:foo---foo
```

File: Lib/test/test_ntpath.py [hunk 1]
```diff
@@ -866,18 +866,25 @@ def test_commonpath(self):
...
         self.assertRaises(ValueError, ntpath.commonpath, [])
         self.assertRaises(ValueError, ntpath.commonpath, iter([]))
-        check_error(ValueError, ['C:\\Program Files', 'Program Files'])
-        check_error(ValueError, ['C:\\Program Files', 'C:Program Files'])
-        check_error(ValueError, ['\\Program Files', 'Program Files'])
-        check_error(ValueError, ['Program Files', 'C:\\Program Files'])
+
+        # gh-117381: Logical error messages, feel free to change drive error
+        # to mix error
+        check_error(ValueError, "Paths don't have the same drive", ['C:\\Program Files', '\\Program Files'])
```
Comment (serhiy-storchaka) [inline Lib/test/test_ntpath.py]: For all tests, also check the arguments in a different order. The error message should be the same.

The patch can have a drive or not, be rooted or not, so there are 4 different kinds of paths. It gives 16 combination of pairs, 4 of them are valid, -- there should be 12 error tests.

File: Lib/test/test_ntpath.py [hunk 2]
```diff
@@ -905,7 +912,8 @@ def check_error(exc, paths):
         check(['c:/program files/bar', 'C:\\Program Files\\Foo'],
               'c:\\program files')
 
-        check_error(ValueError, ['C:\\Program Files', 'D:\\Program Files'])
+        check_error(ValueError, "Paths don't have the same drive", ['C:\\Program Files', 'D:\\Program Files'])
```
Comment (serhiy-storchaka) [inline Lib/test/test_ntpath.py]: Please add also tests for `['C:\\Program Files', 'D:Program Files']` and `['C:Program Files', 'D:\\Program Files']`.

And please break too long lines with `check_error`.

File: Lib/test/test_ntpath.py (hunk 1 above)
Comment (nineteendo) [reply Lib/test/test_ntpath.py]: I've updated `check_error()` to also test the reversed list. I have a schema of all invalid pairs above.

File: Lib/test/test_ntpath.py (hunk 2 above)
Comment (nineteendo) [reply Lib/test/test_ntpath.py]: The first one was already tested above, which I now moved down. I have now modified the tests to use foo, bar & baz to make them easier to read.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**